src/main/java/edu/hm/hafner/analysis/parser/EmbeddedEngineerParser.java

```diff
@@ -40,28 +45,47 @@ public Report parse(final ReaderFactory reader) throws ParsingException {
         }
     }
 
+    private List<String> readLogInfo(final Iterator<String> lineIterator) {
+        List<String> logInfo = new ArrayList<>();
+
+        while (lineIterator.hasNext()) {
+            String currentLine = lineIterator.next();
+            Matcher logBeginningMatcher = LOG_BEGINNING_PATTERN.matcher(currentLine);
+            if (logBeginningMatcher.matches()) {
+                logInfo.add(currentLine);
+            }
+            else {
+                int lastIdx = logInfo.size() - 1;
+                StringBuilder multiLineInfo = new StringBuilder(logInfo.get(lastIdx));
+                multiLineInfo.append(" ").append(currentLine);
+                logInfo.set(lastIdx, multiLineInfo.toString());
+            }
+        }
+        return logInfo;
```

Comment:
I'm not sure how big your logfiles are getting, but you are basically defeating the purpose of the stream by reading everything into memory. Other parsers that want to join additional lines use `LookaheadStream`. See GccParser or MavenParser as an example.

Reply:
thank you for the suggestion.
implementation has been changed to use `LookaheadStream`.